openwisp_controller/config/base/vpn.py

```diff
@@ -710,14 +710,4 @@ def _auto_vxlan(self):
     def _auto_ip(self):
         if not self.vpn.subnet:
             return
-        if self.vpn.subnet.subnetdivisionrule_set.filter(
-            organization_id=self.config.device.organization,
-            type=(
-                'openwisp_controller.subnet_division.rule_types.'
-                'vpn.VpnSubnetDivisionRuleType'
-            ),
-        ).exists():
-            # Do not assign IP here if the VPN has a subnet with a subnet
-            # rule for the organization of the device
-            return
```

Comment:
also here, instead of monkey patching, we can leave room for calling a method which would stop the process, eg:

```python
register_auto_ip_stopper(func)
```

```python
for stopper_func in self._stopper_funcs:
    if stopper_func():
        return
```